lib/tty/file.rb

```diff
@@ -309,10 +319,11 @@ def copy_file(source_path, *args, context: nil, force: false, skip: false,
         content = block[content] if block
         content
       end
-      return unless preserve
+      return file_dest unless preserve
```

Comment:
We need a test case that confirms this behaviour so that it stays true in future releases.